tweets/tests.py

```diff
@@ -1,40 +1,145 @@
 
+from .models import Tweet
+
 User = get_user_model()
 
 
 class TestHomeView(TestCase):
+
     def setUp(self):
         self.user = User.objects.create_user(username="testuser", password="testpassword")
         self.client.login(username="testuser", password="testpassword")
+        self.url = reverse("tweets:home")
 
     def test_success_get(self):
-        response = self.client.get(reverse("tweets:home"))
+
+        response = self.client.get(self.url)
+
         self.assertEqual(response.status_code, 200)
+        # contextに含まれているツイート = DBに保存されているツイートか
+        self.assertEqual(list(response.context["tweets"]), list(Tweet.objects.all()))
 
 
-# class TestTweetCreateView(TestCase):
-#     def test_success_get(self):
+class TestTweetCreateView(TestCase):
 
-#     def test_success_post(self):
+    def setUp(self):
+        self.user = User.objects.create_user(username="testuser", password="testpassword")
+        self.client.login(username="testuser", password="testpassword")
+        self.url = reverse("tweets:create")
 
-#     def test_failure_post_with_empty_content(self):
+    def test_success_get(self):
 
-#     def test_failure_post_with_too_long_content(self):
+        response = self.client.get(self.url)
 
+        self.assertEqual(response.status_code, 200)
 
-# class TestTweetDetailView(TestCase):
-#     def test_success_get(self):
+    def test_success_post(self):
 
+        valid_form_data = {"content": "This is a test tweet"}
+        response = self.client.post(self.url, valid_form_data)
 
-# class TestTweetDeleteView(TestCase):
-#     def test_success_post(self):
+        self.assertRedirects(
+            response,
+            expected_url=reverse(settings.LOGIN_REDIRECT_URL),
+            status_code=302,
+            target_status_code=200,
+        )
+        # DBにデータが追加されているか
+        self.assertEqual(Tweet.objects.all().count(), 1)
+        # 追加されたデータのcontent = 送信されたcontentか？
+        self.assertEqual(Tweet.objects.last().content, valid_form_data["content"])
 
-#     def test_failure_post_with_not_exist_tweet(self):
+    def test_failure_post_with_empty_content(self):
+
+        empty_form_data = {"content": ""}
+        response = self.client.post(self.url, empty_form_data)
+        form = response.context["form"]
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn("このフィールドは必須です。", form.errors["content"])
+        self.assertEqual(Tweet.objects.all().count(), 0)
+
+    def test_failure_post_with_too_long_content(self):
+
+        # max lengh text(280)を超えるtextを用意
+        too_long_text = ""
+        while len(too_long_text) < 290:
+            too_long_text += "too long"
+
+        too_long_form_data = {"content": too_long_text}
+        response = self.client.post(self.url, too_long_form_data)
+        form = response.context["form"]
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn("この値は 280 文字以下でなければなりません( 296 文字になっています)。", form.errors["content"])

```

Comment:
69-71行目で296文字になることが不明瞭なので、`too_long_text`の生成方法を変えた方が良いかもしれません(エラーメッセージには`296文字`とベタ書きしているため)。

```suggestion
        text_length = 296
        too_long_text = "x" * text_length

        too_long_form_data = {"content": too_long_text}
        response = self.client.post(self.url, too_long_form_data)
        form = response.context["form"]

        self.assertEqual(response.status_code, 200)
        self.assertIn(f"この値は 280 文字以下でなければなりません( {text_length} 文字になっています)。", form.errors["content"])
```

Reply:
確かに提案していただいたコードの方が読みやすいですね。